# Profesionalismo en el Ámbito Actuarial
## TAREA 2
## Hernández Martínez Oscar Gerardo

### 1. ¿Es información Confiable, Homogénea y Suficiente?

Sí es información confiable, puesto que son datos reales de una compañía de seguros.

## Carga de librerías

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import math

## Carga de datos

In [2]:
p = pd.read_csv("POLIZAS.CSV", low_memory=False)
r = pd.read_csv("RECLAMACIONES.CSV",low_memory=False)

In [3]:
p

poliza  renovacion  ini_vigpol  fin_vigpol     ini_vig  \
0        100000002822           5  08/01/2004  08/01/2005  08/01/2004   
1        100000002822           5  08/01/2004  08/01/2005  08/01/2004   
2        100000002822           5  08/01/2004  08/01/2005  08/01/2004   
3        100000002822           5  08/01/2004  08/01/2005  08/01/2004   
4        100000002822           5  08/01/2004  08/01/2005  08/01/2004   
...               ...         ...         ...         ...         ...   
1968154         32488          11  05/01/2016  04/30/2017  05/01/2016   
1968155         32489          11  05/01/2016  04/30/2017  05/01/2016   
1968156         32494          13  05/01/2016  04/30/2017  05/01/2016   
1968157         32495          13  05/01/2016  04/30/2017  05/01/2016   
1968158         42412           2  05/01/2016  04/30/2017  05/01/2016   

            fin_vig  prima_neta  ramo  
0        08/01/2005     -561.66    37  
1        08/01/2005     2246.62    37  
2        08/01/2005     -561.66    37  
3        08/01/2005     2246.62    37  
4        08/01/2005     -561.66    37  
...             ...         ...   ...  
1968154  04/30/2017    47699.28    37  
1968155  04/30/2017    19051.32    37  
1968156  04/30/2017    88129.32    37  
1968157  04/30/2017    21582.24    37  
1968158  04/30/2017    15518.16    37  

[1968159 rows x 8 columns]

In [4]:
r

ini_vig     fin_vig  fecha_pago  fech_ocurr  importe  ramo
0        03/08/2010  03/08/2011  09/06/2011  14/01/2011   469.35    37
1        03/12/2009  03/12/2010  30/11/2010  14/06/2010   115.96    37
2        03/12/2009  03/12/2010  30/11/2010  14/06/2010    34.70    37
3        03/12/2009  03/12/2010  30/11/2010  14/06/2010    87.66    37
4        03/08/2010  03/08/2011  09/06/2011  14/01/2011    78.32    37
...             ...         ...         ...         ...      ...   ...
4733650  08/04/2016         NaN  22/04/2016  22/04/2016   200.00    39
4733651  08/04/2016         NaN  22/04/2016  22/04/2016   330.00    39
4733652  01/04/2016         NaN  22/04/2016  22/04/2016   200.00    39
4733653  11/04/2016         NaN  22/04/2016  22/04/2016   330.00    39
4733654  11/04/2016         NaN  22/04/2016  22/04/2016   330.00    39

[4733655 rows x 6 columns]

In [5]:
len(p['poliza'].unique())

51010

In [6]:
p['ramo'].value_counts()

37    1531804
39     436354
34          1
Name: ramo, dtype: int64

In [18]:
r['ramo'].value_counts()

39    3059892
37    1673761
36          2
Name: ramo, dtype: int64

Nos encontramos con que la base consta de 51010 pólizas diferentes, por lo que se puede afirmar que es suficiente. 

Para el ramo 37 (Salud Individual) contamos con la información suficiente y homogénea, por lo tanto, se puede realizar el cálculo de la prima de tarifa.

La información para el ramo 34 (Gastos Médicos Individuales) no es suficiente por lo que no se puede determinar la prima de tarifa.

Para el ramo 39 (Salud Colectivo), contamos con suficiente información, sin embargo, al no conocer los datos de las unidades expuestas, no se puede determinar si la información es homogénea, por lo que no se puede realizar el cálculo de la prima de tarifa. 

Por lo tanto, la información disponible no permite realizar el cálculo de la prima de tarifa a corto plazo ni a largo plazo.

Además, al no conocer el nombre de la aseguradora, no se puede saber el margen de utilidad que maneja, dato necesario para el cálculo.

In [20]:
pol = p.groupby('ramo').sum()
pol

poliza  renovacion    prima_neta
ramo                                              
34                     1           0  1.440360e+03
37    149190769800911808     4027570  1.873547e+09
39     80428823261819456     1273567  2.261303e+09

In [21]:
rec = r.groupby('ramo').sum()

importe
ramo              
36    0.000000e+00
37    8.761403e+08
39    1.314700e+09

Algo interesante que se puede apreciar, es que la base de reclamaciones cuenta con un ramo que no aparece en la base de pólizas, (ramo 36, Gastos Médicos Colectivo). Sin embargo, este ramo cuenta con dos registros y al hacer la suma, nos topamos con que el importe es cero, lo cual puede significar un error y corrección o bien, puede hacernos dudar sobre la veracidad de la información.


<img src=Imagen1.png>

<img src=Imagen2.png>

<img src=Imagen3.png>

Como se puede apreciar en las gráficas realizadas con Power BI, observamos que existe una deficiencia de las primas del ramo 37 y el ramo 34. 